### PR TITLE
Mergesort for lag transfer

### PR DESCRIPTION
**Pull request by ywx649999311**
## Change Description
Replace jax.argsort with custom merge sort to speed likelihood evaluation in lag inference. Now the time spend on sorting per likelihood evaluation should be negligible compared to GP log_prob.

## Code Quality
- [x] I have read the Contribution Guide and agree to the Code of Conduct
- [x] My code follows the code style of this project
- [x] My code builds (or compiles) cleanly without any errors or warnings
- [x] My code contains relevant comments and necessary documentation

### CODE REVIEW

**Comment by ywx649999311** [comment]
TODO:
1. ~~update the syntax for MCMC, such that the model itself is not jitted, only the sampling function~~
2. ~~add test for MCMC~~
3. ~~update tutorials for syntax~~
4. ~~how to deal with band labels not consecutive~~

**Review by camposandro — APPROVED** [review]
Looks good! 

Quick note from our in-person conversation. Since ordering of repeated timestamps does not really matter, it might still be worth it to change the previous [lag_transform](https://github.com/LSST-AGN-Variability/EzTaoX/blob/45e9c2d5d07e8bc706d5399717cb024dd89fe530/src/eztaox/models.py#L178) to perform an unstable sort instead: `jnp.argsort(new_t, stable=False)`. It has a significant speedup, though it seems that merge sort is still faster.
